### PR TITLE
WT-6442 Add statistics and logging to track out-of-order handling

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
Added some statistics to track certain conditions in our out-of-order handling that might be interesting to us as well as some verbose logging to help us debug.

The log lines look like this:
```
[1593047094:735561][14719:0x7f25f64fd700], test_timestamp20.test_timestamp20.test_timestamp20, file:test_timestamp20.wt, eviction-server: [WT_VERB_TIMESTAMP] fixing out-of-order updates during insertion; start_ts=(0, 30), durable_start_ts=(0, 30), min_insert_ts=(0, 5)
[1593047094:740942][14719:0x7f25f64fd700], test_timestamp20.test_timestamp20.test_timestamp20, file:WiredTigerHS.wt, eviction-server: [WT_VERB_TIMESTAMP] fixing existing out-of-order updates by moving them; start_ts=(0, 10), durable_start_ts=(0, 10), stop_ts=(0, 20), durable_stop_ts=(0, 20), new_ts=(0, 5)
```

### CODE REVIEW

**Comment by tetsuo-cpp** [comment]
@agorrod @kommiharibabu 
This one is just stats and debugging information so it's not urgent.

File: dist/stat_data.py [hunk 1]
```diff
@@ -303,6 +303,9 @@ def __init__(self, name, desc, flags=''):
     CacheStat('cache_hs_key_truncate_mix_ts_restart', 'history store key truncation calls that returned restart'),
     CacheStat('cache_hs_ondisk', 'history store table on-disk size', 'no_clear,no_scale,size'),
     CacheStat('cache_hs_ondisk_max', 'history store table max on-disk size', 'no_clear,no_scale,size'),
+    CacheStat('cache_hs_order_fixup_insert', 'history store table out-of-order updates that were fixed up during insertion'),
+    CacheStat('cache_hs_order_fixup_move', 'history store table out-of-order updates that were fixed up by moving existing records'),
+    CacheStat('cache_hs_order_resolved', 'history store table out-of-order resolved updates that lose their durable timestamp'),
```
Comment (kommiharibabu) [inline dist/stat_data.py:308]: Is it possible to add durable timestamp lost in the stat name? How about `cache_hs_order_lose_durable_timestamp`.

Comment (tetsuo-cpp) [reply dist/stat_data.py]: ea0294d

Thanks. That is a better name.

**Comment by tetsuo-cpp** [comment]
@agorrod 
Is it ok to merge this? I saw you commented to give it permission to go in during code freeze.

**Comment by agorrod** [comment]
@tetsuo-cpp yes - as long as it passes MongoDB testing cleanly I'm good for this to go in.

**Pull request merged**